.circleci/config.yml

```
@@ -437,6 +432,94 @@ jobs:
     steps:
       - run-build:
           number-vcpu: 8
+
+  tarball-assembly:
+    machine:
+      image: ubuntu-1604:201903-01
+      docker_layer_caching: true
+    environment:
+      DOCKER_CLI_EXPERIMENTAL: enabled
+    parameters:
+      number-vcpu:
+        default: 2
+        type: integer
+      vaadin-javamaxmem:
+        default: 1g
+        type: string
+    steps:
+      - attach_workspace:
+          at: ~/
+      - run:
+          name: multiarch/qemu-user-static
+          command: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+      - run:
+          name: Install Docker buildx
+          command: |
+            sudo wget https://github.com/docker/buildx/releases/download/v0.3.1/buildx-v0.3.1.linux-amd64 -O /usr/local/bin/docker-buildx
+            sudo chmod a+x /usr/local/bin/docker-buildx
+            sudo systemctl restart docker
+      - dockerhub-login
+      - run:
+          name: Assemble tarballs
+          # Install fake makensis to satisfy the assemble dependency
+          command: |
+            sudo cp .circleci/scripts/makensis.py /usr/local/bin/makensis
+            ./assemble.pl -Dopennms.home=/opt/opennms \
+                          -DskipTests \
+                          -DupdatePolicy=never
+      - run:
+          name: Collect tarball artifacts
+          command: |
+            mkdir target/tarballs
+            find ./target -name "*.tar.gz" -type f -not -iname '*source*' -exec cp {} ./target/tarballs/opennms.tar.gz \;
+            find ./opennms-assemblies/minion/target -name "*.tar.gz" -type f -not -iname '*source*' -exec cp {} ./target/tarballs/minion.tar.gz \;
+            find ./opennms-assemblies/sentinel/target -name "*.tar.gz" -type f -not -iname '*source*' -exec cp {} ./target/tarballs/sentinel.tar.gz \;
+      - run:
+          name: Build Minion OCI
+          command: |
+            cd opennms-container/minion
+            make DOCKER_TAG="minion:latest" BUILD_NUMBER="${CIRCLE_BUILD_NUM}" BUILD_URL="${CIRCLE_BUILD_URL}" BUILD_BRANCH="${CIRCLE_BRANCH}"
+            case "${CIRCLE_BRANCH}" in
+              master)
```

```suggestion
              "master" | "develop")
```

It looks like the code blocks here are the same. We can de-duplicate them with an OR for the case statement?